### ISSUE TITLE
Issues with Mysterium node

### ISSUE THREAD

**Issue by mkfox14** [mysteriumnetwork/node#5047]
The following error occurs frequently in mysterium-node.log and I don't seem to accumulate MYST as I should:

2022-04-20T01:12:11.343 INF session/pingpong/hermes_promise_settler.go:282 > Received hermes promise for {"0xb146ffca128051634c6f0b3fe43f2962ee21a06c"}
2022-04-20T01:12:11.344 INF session/pingpong/hermes_promise_settler.go:309 > Hermes "0xDe82990405aCc36B4Fd53c94A24D1010fcc1F83d" promise state updated for provider {"0xb146ffca128051634c6f0b3fe43f2962ee21a06c"}
2022-04-20T01:12:11.344 INF session/pingpong/hermes_channel_repository.go:343 > Loaded state for provider {"0xb146ffca128051634c6f0b3fe43f2962ee21a06c"}, hermesID "0xDe82990405aCc36B4Fd53c94A24D1010fcc1F83d": balance 0, available balance 0, unsettled balance 35716926273200024
2022-04-20T01:13:02.289 INF session/pingpong/hermes_promise_handler.go:422 > Recovering R...
2022-04-20T01:13:02.289 INF session/pingpong/hermes_promise_handler.go:439 > R recovered, will reveal...
2022-04-20T01:13:02.667 WRN session/pingpong/invoice_tracker.go:565 > unknown hermes error encountered, will retry error="hermes request promise error: could not reveal R: could not reveal R for hermes: r recovery required"

I am also to unable to withdraw to my wallet -- the same error "could not reveal R for hermes: r recovery required" errors occurs.

Node version is 1.6.3 on Ubuntu 20.04.  I believe the error may have started occurring *after* I moved my node from a RaspberryPi to Ubuntu Linux after experiencing file system corruption on the Pi around 6 weeks ago.  (I simply stopped the mysterium-node processes and copied the /var/lib/mysterium-node directory from the Pi to Ubuntu.)

Identity is 0xb146ffca128051634c6f0b3fe43f2962ee21a06c.

**Comment by etherunit** [comment]
@mkfox14 Can you please try to update your node, then reboot it and try again?

**Comment by mkfox14** [comment]
@etherunit - I have upgraded from my node from 1.6.3 to 1.6.4 and rebooted.  (Also a Linux kernel update from 5.4.0-104 to 5.4.0-109 if that has any relevance.)  I waited for a session to close, and it looks like a similar error:

2022-04-20T15:19:51.490 DBG session/pingpong/invoice_messaging.go:57 > Sending P2P message to "p2p-payment-invoice": AgreementID:"60416974611365757148414901816356140100629840219274689166757279006234330527355"  AgreementTotal:"126913062806730"  TransactorFee:"0"  Hashlock:"80169ae60eafe7322a1dfdcf071ac0b357bdcde668b4479d9e49edac49915698"  Provider:"0xb146ffca128051634c6f0b3fe43f2962ee21a06c"  ChainID:137
2022-04-20T15:19:51.744 DBG core/service/subscription.go:148         > Received P2P message for "p2p-payment-message": Promise:{ChannelID:"\xf6ڔR\xea\xfa`\x17oށ-\xfbj\xf7\x13\xf1W\xf2\xbb"  Amount:"209513911670314093"  Fee:"0"  Hashlock:"\x80\x16\x9a\xe6\x0e\xaf\xe72*\x1d\xfd\xcf\x07\x1a\xc0\xb3W\xbd\xcd\xe6h\xb4G\x9d\x9eI\xed\xacI\x91V\x98"  ChainID:137  Signature:"\x92\xcf\xf6ߚ\x9d]\x87e\xccUq\xd5\xe9\x81\x01\x97\xe9ɏ\xaa\x98\xee\xcf\x13_\x8d\x03gi\xbd\xf9\x1428\xf0z4\xc8ls\x1eOW\x04\xe8\xeb\x0b\x004#u:K\x8aI\xb2\xa43\x9cSwg6\x1b"}  AgreementID:"60416974611365757148414901816356140100629840219274689166757279006234330527355"  AgreementTotal:"126913062806730"  Provider:"0xb146ffca128051634c6f0b3fe43f2962ee21a06c"  Signature:"fe49b3fb61641372720967ead30c0498d4200f3162e0f044137c4b15a61fb6f365105d836bf0644e0bad6e50a047287487becf7929c6905922a7021c0ee5663a1c"  HermesID:"0xa62a2A75949d25e17C6F08a7818e7bE97c18a8d2"  ChainID:137
2022-04-20T15:19:52.331 INF session/pingpong/hermes_promise_handler.go:422 > Recovering R...
2022-04-20T15:19:52.332 INF session/pingpong/hermes_promise_handler.go:439 > R recovered, will reveal...
2022-04-20T15:19:52.692 WRN session/pingpong/invoice_tracker.go:565  > unknown hermes error encountered, will retry error="hermes request promise error: could not reveal R: could not reveal R for hermes: r recovery required"


If I attempt a withdrawal, here is the error:

2022-04-20T15:22:57.269 INF session/pingpong/hermes_promise_settler.go:499 > Marked provider {0xb146ffca128051634c6f0b3fe43f2962ee21a06c} as requesting settlement
2022-04-20T15:22:58.538 INF session/pingpong/hermes_promise_handler.go:422 > Recovering R...
2022-04-20T15:22:58.539 INF session/pingpong/hermes_promise_handler.go:439 > R recovered, will reveal...
2022-04-20T15:22:58.903 ERR tequilapi/endpoints/transactor.go:506    > Withdrawal failed error="could not call hermes pay and settle:hermes request promise error: could not reveal R: could not reveal R for hermes: r recovery required" amount=50000000000000000000 beneficiary=0x0eAE19c8A6B313dE5f47F2BbC0D2Fa707aB6bE83 from_chain_id=137 hermes_id=0xa62a2A75949d25e17C6F08a7818e7bE97c18a8d2 provider_id=0xb146ffca128051634c6f0b3fe43f2962ee21a06c to_chain_id=137

**Comment by etherunit** [comment]
@Guillembonet Any hints there?

**Comment by Guillembonet** [comment]
yep, it seems like that R you need is lost so you are stuck there, there should be a way to recover from this, I will take a look and let you know!

**Comment by Guillembonet** [comment]
also, maybe you'll need to try to withdraw/wait for a promise request twice to get it unstuck

**Comment by mkfox14** [comment]
The hotfix has resolved the issue!  I have been able to withdraw to my wallet.  The log now says "R recovered successfully" where it was failing before.  Thanks for the assistance!
